packages/wonder-blocks-modal/components/modal-content-pane.js

```diff
@@ -63,12 +66,18 @@ export default class ModalContentPane extends React.Component<Props> {
                 <ModalContent>{content}</ModalContent>
             );
 
-        if (mainContent && (titleBar || footer || header)) {
+        if (mainContent) {
             mainContent = React.cloneElement(mainContent, {
+                scrollOverflow,
                 header: header || mainContent.props.header,
                 style: [
                     titleBar && styles.hasTitleBar,
                     footer && styles.hasFooter,
+                    showCloseButton &&
+                        !titleBar &&
+                        ((gridSize) =>
+                            gridSize === "small" &&
+                            styles.smallWithCloseButton),
```

Comment:
Are we actually intending people to use the grid within the modals?  If not, we should probably just you media queries instead.

Reply:
It might also be worth adding a comment as to why we have to inject `mainContent`'s style generally speaking.

Reply:
I don't think we should be using media queries at all - the grid replaces it entirely. I could renamed the main `<Grid>` component to be `<MediaLayout>` (or something) as that's really what it is - it's implementing a set of media queries and layouts that will be followed - once the content is wrapped with that there's no longer any reason to be using media queries.

Reply:
This is a pretty big shift in thinking.  I think calling it `MediaLayout` would help communicate that.  And then a grid layout is implemented using a combination of rows and cells and that layout changes in response to changes to the media that `MediaLayout` then feeds down to descendant components via context.